.gitlab-ci.yml

```diff
@@ -24,70 +22,59 @@ stages:
 - cleanup
 
 before_script:
-  # See https://docs.gitlab.com/ee/ci/ssh_keys/README.html
-  - eval $(ssh-agent -s) && ssh-add <(echo "$SSH_PRIVATE_KEY")
-  - mkdir -p $HOME/.ssh && echo "StrictHostKeyChecking no" >> "$HOME/.ssh/config"
-  - export PATH=$PATH:$HOME/bin:$HOME/terminus/bin
-  - export PR_ENV=${CI_MERGE_REQUEST_IID:+pr-$CI_MERGE_REQUEST_IID}
-  - export TERMINUS_ENV=${PR_ENV:-dev}
-  - export BEHAT_ADMIN_PASSWORD=$(openssl rand -base64 24)
-  - terminus auth:login --machine-token="$TERMINUS_TOKEN"
-  - git config --global user.email "$GITLAB_USER_EMAIL"
-  - git config --global user.name "Gitlab CI"
+- export PATH="$PATH:$CI_PROJECT_DIR/.ci/scripts"
+- export BASH_ENV="$HOME/.envvarsrc"
+- /build-tools-ci/scripts/set-environment
+- source $BASH_ENV
+# See https://docs.gitlab.com/ee/ci/ssh_keys/README.html
+- eval $(ssh-agent -s) && ssh-add <(echo "$SSH_PRIVATE_KEY")
+- terminus -n auth:login --machine-token="$TERMINUS_TOKEN"
 
-deploy:multidev:
+deploy:
   stage: build
   environment:
-    name: review/pr-$CI_MERGE_REQUEST_IID
-    url: https://pr-$CI_MERGE_REQUEST_IID-$TERMINUS_SITE.pantheonsite.io/
+    name: deploy/$TERMINUS_ENV
+    url: https://$TERMINUS_ENV-$TERMINUS_SITE.pantheonsite.io/
   script:
-    - terminus build:env:delete:pr -n "$TERMINUS_SITE" --yes
-    - composer build-assets
-    - terminus env:wake -n "$TERMINUS_SITE.dev"
-    - terminus build-env:create -n "$TERMINUS_SITE.dev" "$TERMINUS_ENV" --yes
-  only: 
-    - merge_requests
-
-deploy:dev:
-  stage: build
-  environment:
-    name: dev
-    url: https://dev-$TERMINUS_SITE.pantheonsite.io/
-  script:
-    - composer build-assets
-    - terminus build-env:push-code $TERMINUS_SITE.dev
-    - terminus build:env:delete:pr -n "$TERMINUS_SITE" --yes
+  - echo "Target site.env is $TERMINUS_SITE.$TERMINUS_ENV"
+  - 01-prepare
+  - composer -n build-assets
+  - 02-init-site-under-test-clone-existing
   only:
-    - master
+  - merge_requests
+  - master
 
 test:code_sniff_unit_test:
   stage: test
   script:
-    - composer install --no-ansi --no-interaction --optimize-autoloader --no-progress
-    - rm -rf web
-    - git checkout web
-    - composer -n code-sniff
-    - composer -n unit-test
+  - composer install --no-ansi --no-interaction --optimize-autoloader --no-progress
+  - rm -rf web
+  - git checkout -- web
```

Comment:
Resetting `web` was also necessary. This isn't done in the `example-drops-8-composer` version, though I don't think it would hurt there.

The previous workaround to not running code sniffing/unit testing/linting on contrib code was to run `composer install` in an earlier job and persist `vendor` to this job, leaving `web` intact without contrib items.

Reply:
Removing and resetting `web` is not ideal, and I'd rather avoid it if possible. I'm not sure if there is or is not a better way, though.

Reply:
The alternative was to run another job and persist `vendor` with just dev dependencies in it but not all CI providers support such persistence (GitLab does not).

Reply:
The problem as I understand it is that `code-sniff` will fail if run on a site with contrib plugins with coding errors. I see that as a shortcoming in the configuration of `code-sniff` and `rm`ing web as a rough workaround. I don't want to hold up this PR on that detail because it's not a change. I would like developers to be able to run `code-sniff` locally though.